src/providers/data.rs

```diff
@@ -294,19 +293,13 @@ pub trait Format: Sized {
 
     /// Parses `string` as the data format `Self` as a `T` or returns an error
     /// if the `string` is an invalid `T`.
-    fn from_str<'de, T: DeserializeOwned>(string: &'de str) -> Result<T, Self::Error>;
+    fn from_str<T: DeserializeOwned>(string: &str) -> Result<T, Self::Error>;
 
     /// Parses the file at `path` as the data format `Self` as a `T` or returns
     /// an error if the `string` is an invalid `T`. The default implementation
     /// calls [`Format::from_str()`] with the contents of the file.
     fn from_path<T: DeserializeOwned>(path: &Path) -> Result<T, Self::Error> {
-        let mut source = String::new();
-        std::fs::File::open(path)
-            .map(|file| BufReader::new(file))
-            .and_then(|mut reader| reader.read_to_string(&mut source))
-            .map_err(|e| de::Error::custom(e))?;
-
-        Self::from_str(&source)
+        Self::from_str(&std::fs::read_to_string(path).map_err(de::Error::custom)?)
```

Comment:
While this is a lot simpler, we could keep the source variable part to make it more readable, like `let source = std::fs::read_to_string(path).map_err(de::Error::custom)?; Self::from_str(&source)`

Reply:
Let's do that.